(DOCSP-27757) Update Configure a Realm pages to reflect latest `@realm/react` version (#2900)
mongodb/docs-realm: source/sdk/react-native/sync-data/configure-a-synced-realm.txt

   You must configure Flexible Sync in the backend before you can use it with your
   client application.
#. :ref:`Initialize the App client <react-native-connect-to-mongodb-realm-backend-app>`.
#. :ref:`Authenticate a user <react-native-authenticate-users>` in your client
   project.

.. _react-native-flexible-sync-open-realm:

Configure a Synced Realm
------------------------

> You configre a a synced realm using the providers from ``@realm/react``.

Reviewer: ```suggestion
Configure a synced realm using the providers from ``@realm/react``.
```

Author: I love to `configre` things. Sounds fancier than just configure. 😆 
Thank you for catching this mess.

   but you can also set up subscriptions in ``RealmProvider`` child components.

This is how you nest providers:

.. literalinclude:: /examples/generated/react-native/ts/configure-realm-sync.test.snippet.configure-realm-sync-full.tsx
   :language: javascript

.. note:: Partition-Based Sync

   This page covers Flexible Sync realms. Flexible Sync is the preferred mode for
   new apps that use :ref:`Atlas Device Sync <sync>`. For information about realms
   using the older Partition-Based Sync, refer to
   :ref:`Partition-Based Sync <react-native-partition-realms>`.

.. _react-native-sync-app-provider:

AppProvider
~~~~~~~~~~~

To set up your App client, pass the App ID string to the ``id`` prop of
``AppProvider``. Wrap any components that need to access the App with the
``AppProvider``.

.. literalinclude:: /examples/generated/react-native/ts/app-provider.test.snippet.app-provider.tsx
   :language: typescript

You can find more information about ``AppProvider`` on the
:ref:`Connect To Atlas App Services page <react-native-app-client-configuration>`.

.. _react-native-sync-user-provider:

UserProvider
~~~~~~~~~~~~

> Configure user authentication with ``AppProvider`` and ``UserProvider``. Then

Reviewer: ```suggestion
Configure user authentication with ``AppProvider`` and ``UserProvider``. Then,
```

that exposes a realm to its child components. You configure your realm by
passing props to ``RealmProvider``.

When ``RealmProvider`` is rendered, it opens the realm. This means that the
provider renders successfully or its child components can't access the realm.

To configure a synced realm:

#. Import :ref:`providers <react-native-sync-providers-hooks>` from
   ``@realm/react``.
#. Configure ``AppProvider``.
#. Configure ``UserProvider`` and nest it within ``AppProvider``.
#. Pass your object models to ``RealmProvider``'s ``schema`` prop.
#. Create a :js-sdk:`SyncConfiguration <Realm.App.Sync.html#~SyncConfiguration>`
   object.
#. Pass your ``SyncConfiguration`` object to the ``sync`` prop or add the object
   in-line.
#. Set up initial subscriptions or create a new subscription in
   ``RealmProvider`` child components.
#. Add other :js-sdk:`Configuration object <Realm.html#~Configuration>` 
   properties as props to ``RealmProvider`` to further configure your realm.

.. literalinclude:: /examples/generated/react-native/ts/configure-realm-sync.test.snippet.configure-realm-sync-full.tsx
   :language: typescript

>    :emphasize-lines: 9-19

Reviewer: I think the highlighting here is off. If you're trying to highlight `RealmProvider`, we're missing the opening line.
```suggestion
   :emphasize-lines: 8-19
```
